src/tokens/ERC5095.sol

```diff
@@ -196,17 +220,95 @@ contract ERC5095 is ERC20Permit, IERC5095 {
             shares
```

Comment:
Ive realized you just need to input `m` here and the marketplace's slippage controls should take care of things.

(Meaning we dont need to have the block above on line 207)